hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/sink/StreamWriteFunction.java

```diff
@@ -448,7 +448,8 @@ private boolean flushBucket(DataBucket bucket) {
   }
 
   @SuppressWarnings("unchecked, rawtypes")
-  private void flushRemaining(boolean endInput) {
+  @VisibleForTesting
+  protected void flushRemaining(boolean endInput) {
```

Comment:
Do we need this?

Reply:
Yeap, so we can override this function in `BucketStreamWriteTestFunction`.